webapp/src/components/news-list/NewsList.test.js

```diff
@@ -0,0 +1,162 @@
+import {describe, expect, it} from "@jest/globals";
+import {mount} from "@vue/test-utils";
+import NewsList from "@/components/news-list/NewsList";
+import NewsItem from "@/components/news-item/NewsItem";
+import NewsForm from "@/components/news-form/NewsForm";
+
+require('regenerator-runtime/runtime');
+
+  describe('NewsList', () => {
+    it('should show the message "The list is empty :(" when there are no elements in the list', () => {
+      const wrapper = mount(NewsList, {
+          data() {
+              return {
+                  newsItems: [],
+              }
+          }
+      })
+      const paragraph = wrapper.find("h2");
+
+      expect(paragraph.text()).toEqual("The list is empty :(");
+      wrapper.destroy();
+  })
+
+  it("should show the elemens when they are present in the list", () => {
+      const wrapper = mount(NewsList, {
+          data() {
+              return {
+                  newsItems: [
+                      {title: "Title_1", votes: 0},
+                      {title: "Title_2", votes: 0}
+                  ]
+              }
+          }
+      });
+
+      let children = wrapper.findAllComponents(NewsItem);
+      expect(children.length).toEqual(2);
+      wrapper.destroy();
+  })
+
+  it("should switch the sorting order whenever the button is clicked", () => {
+      const wrapper = mount(NewsList, {
+          data() {
+              return {
+                  newsItems: [
+                      {title: "Title_1", votes: 5},
+                      {title: "Title_2", votes: 4}
```

Comment:
If you need to change some options of your test setup but you want to re-use the setup:

```js
let = setupWrapper(options) => {
  const defaults = { propsData: { newsItems: [] } }
  return mount(NewsList, { ...defaults, ...options })
}

// .. later in your tests:
const wrapper = setupWrapper({propsData: { newsItems: [{title: "Title_1", votes: 5}]} })